OutOfSchool/OutOfSchool.BusinessLogic/Services/ExternalExportService.cs

```diff
@@ -0,0 +1,137 @@
+﻿using AutoMapper;
+using OutOfSchool.BusinessLogic.Models;
+using OutOfSchool.BusinessLogic.Models.Exported;
+using OutOfSchool.BusinessLogic.Services.AverageRatings;
+using OutOfSchool.Services.Repository.Api;
+
+namespace OutOfSchool.BusinessLogic.Services;
+
+public class ExternalExportService : IExternalExportService
+{
+    private readonly IProviderRepository providerRepository;
+    private readonly IWorkshopRepository workshopRepository;
+    private readonly IAverageRatingService averageRatingService;
+    private readonly IMapper mapper;
+    private readonly ILogger<ExternalExportService> logger;
+
+    public ExternalExportService(
+        IProviderRepository providerRepository,
+        IWorkshopRepository workshopRepository,
+        IAverageRatingService averageRatingService,
+        IMapper mapper,
+        ILogger<ExternalExportService> logger)
+    {
+        this.providerRepository = providerRepository ?? throw new ArgumentNullException(nameof(providerRepository));
+        this.workshopRepository = workshopRepository ?? throw new ArgumentNullException(nameof(workshopRepository));
+        this.averageRatingService =
+            averageRatingService ?? throw new ArgumentNullException(nameof(averageRatingService));
+        this.mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+        this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+    }
+
+    public async Task<SearchResult<ProviderInfoBaseDto>> GetProviders(DateTime updatedAfter,
+        OffsetFilter offsetFilter)
+    {
+        try
+        {
+            logger.LogInformation("Getting all updated providers started");
+            offsetFilter ??= new OffsetFilter();
+            var providers = await providerRepository
+                .GetAllWithDeleted(updatedAfter, offsetFilter.From, offsetFilter.Size)
+                .ConfigureAwait(false);
+
+            if (providers == null)
+            {
+                logger.LogError("Failed to retrieve updated providers. The provider list is null");
+                return new SearchResult<ProviderInfoBaseDto>();
+            }
```

Comment:
Curious if it is ever a case? Do your repositories return null instead of empty list sometimes?